netbox/resource_netbox_virtual_machine.go

```diff
@@ -76,6 +76,7 @@ func resourceNetboxVirtualMachine() *schema.Resource {
 			"disk_size_gb": {
 				Type:     schema.TypeInt,
 				Optional: true,
```

Comment:
Is this "optional" still needed?

Does it even make sense now that virtual disks exist?

Can I still set a `disk_size_gb` of N for a given VM and it will stay at N until I add a virtual disk, which will then update the `disk_size_gb` accordingly?

Or is `disk_size_gb` now read only in the API as well?

Reply:
> Is this "optional" still needed?

Perhaps `optional` and `computed` are redundant, so I guess it can be removed.

> Does it even make sense now that virtual disks exist?

> Can I still set a disk_size_gb of N for a given VM and it will stay at N until I add a virtual disk, which will then update the disk_size_gb accordingly?

> Or is disk_size_gb now read only in the API as well?

`disk_size_gb` can still be set like before, so I would not remove it and break existing users of this field. But it does not make a lot of sense to use both at the same time. Netbox does enforce these values to be consistent (`disk_size_gb` must be the sum of all disks).

So it's a choice of breaking changes versus a chance of conflicting resources. We could also mark the field deprecated to give users a chance to migrate to the new resources and remove it later.

Reply:
Last bit: Every time a change is made to VMs, we have to inspect the behavior of the `netbox_primary_ip` resource.


We set the primary ip of a VM by getting the VM, setting the primary IP and saving the whole thing back to the VM (in https://github.com/e-breuninger/terraform-provider-netbox/blob/854a0bc56fe2242f8d9545596bf5e8b43aadcad1/netbox/resource_netbox_primary_ip.go#L117 ).

Since the tests run, at least existing code should not break. If netbox maintains consistency on the `Disk` attribute, I see no reason why it would not work once virtual disks come into play.

Agree?